Build failure on stable with non-default features

FYI, looks like the issue with #224 has arisen on stable. `uom` fails to compile on the latest stable version: 1.49.0, with `default-features = false` and the following options: `["f64", "std", "si", "use_serde"]`. I did confirm that going back to the default feature set and just adding "use_serde" works on 1.49.0, however.

Sample of the error listing: 
```
   ::: /Users/me/.cargo/registry/src/github.com-1ecc6299db9ec823/uom-0.30.0/src/system.rs:244:20
    |
244 |           pub struct Quantity<D, U, V>
    |                      -------- required by a bound in this
245 |           where
246 |               D: Dimension + ?Sized,
    |                  --------- required by this bound in `si::Quantity`
    |
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider further restricting type parameter `N`
    |
290 |                 V: crate::num_traits::Num + crate::Conversion<V>, N: Integer
    |                                                                 ^^^^^^^^^^^^

error[E0277]: the trait bound `J: Integer` is not satisfied
```

--- a/src/si/mod.rs
+++ b/src/si/mod.rs
@@ -286,6 +286,13 @@ pub mod marker {
                     V,
                 >
             where
+                L: crate::typenum::Integer,
+                M: crate::typenum::Integer,
+                T: crate::typenum::Integer,
+                I: crate::typenum::Integer,
+                Th: crate::typenum::Integer,
+                N: crate::typenum::Integer,
+                J: crate::typenum::Integer,
                 U: Units<V> + ?Sized,
                 V: crate::num_traits::Num + crate::Conversion<V>,
             {

--- a/src/system.rs
+++ b/src/system.rs
@@ -482,7 +482,8 @@ macro_rules! system {
                     for Quantity<Dl, U, V>
                 where
                     Dl: Dimension + ?Sized,
-                    $(Dl::$symbol: $crate::lib::ops::$AddSubTrait<Dr::$symbol>,)+
+                    $(Dl::$symbol: $crate::lib::ops::$AddSubTrait<Dr::$symbol>,
+                    <Dl::$symbol as $crate::lib::ops::$AddSubTrait<Dr::$symbol>>::Output: $crate::typenum::Integer,)+
                     Dl::Kind: $crate::marker::$MulDivTrait,
                     Dr: Dimension + ?Sized,
                     Dr::Kind: $crate::marker::$MulDivTrait,

